_examples/widget_demos/button_mask/main.go

```diff
@@ -0,0 +1,194 @@
+		//widget.ButtonOpts.Text("Hello, [color=FF00FF]World![/color]", face, &widget.ButtonTextColor{
+		widget.ButtonOpts.Text("Hello!", face, &widget.ButtonTextColor{
+			Idle:  color.NRGBA{255, 255, 255, 255},
+			Hover: color.NRGBA{0, 0, 0, 255},
+		}),
+		//widget.ButtonOpts.TextProcessBBCode(true),
+		// specify that the button's text needs some padding for correct display
+		/*
+			widget.ButtonOpts.TextPadding(widget.Insets{
+				Left:   30,
+				Right:  30,
+				Top:    5,
+				Bottom: 5,
+			}),
+		*/
+
+		// add a handler that reacts to clicking the button
+		widget.ButtonOpts.ClickedHandler(func(args *widget.ButtonClickedEventArgs) {
+			println("button clicked")
+		}),
+
+		// add a handler that reacts to entering the button with the cursor
+		widget.ButtonOpts.CursorEnteredHandler(func(args *widget.ButtonHoverEventArgs) {
+			println("cursor entered button: entered =", args.Entered, "offsetX =", args.OffsetX, "offsetY =", args.OffsetY)
+		}),
+
+		// add a handler that reacts to moving the cursor on the button
+		widget.ButtonOpts.CursorMovedHandler(func(args *widget.ButtonHoverEventArgs) {
+			println("cursor moved on button: entered =", args.Entered, "offsetX =", args.OffsetX, "offsetY =", args.OffsetY, "diffX =", args.DiffX, "diffY =", args.DiffY)
+		}),
+
+		// add a handler that reacts to exiting the button with the cursor
+		widget.ButtonOpts.CursorExitedHandler(func(args *widget.ButtonHoverEventArgs) {
+			println("cursor exited button: entered =", args.Entered, "offsetX =", args.OffsetX, "offsetY =", args.OffsetY)
+		}),
+
+		// Indicate that this button should not be submitted when enter or space are pressed
+		// widget.ButtonOpts.DisableDefaultKeys(),
+	)
+
+	// add the button as a child of the container
+	rootContainer.AddChild(button)
+
+	// construct the UI
+	ui := ebitenui.UI{
+		Container: rootContainer,
+	}
+
+	// Ebiten setup
+	ebiten.SetWindowSize(400, 400)
+	ebiten.SetWindowTitle("Ebiten UI - Buttons with mask")
+
+	game := game{
+		ui:  &ui,
+		btn: button,
+	}
+
+	// run Ebiten main loop
+	err := ebiten.RunGame(&game)
+	if err != nil {
+		log.Println(err)
+	}
+}
+
+// Layout implements Game.
+func (g *game) Layout(outsideWidth int, outsideHeight int) (int, int) {
+	return outsideWidth, outsideHeight
+}
+
+// Update implements Game.
+func (g *game) Update() error {
+	// update the UI
+	g.ui.Update()
+	if inpututil.IsKeyJustPressed(ebiten.KeyB) {
+		g.btn.Click()
+	}
+
+	//Test that you can call Click on the focused widget.
+	if inpututil.IsKeyJustPressed(ebiten.KeyF) {
+		if btn, ok := g.ui.GetFocusedWidget().(*widget.Button); ok {
+			btn.Click()
+		}
+	}
+
+	return nil
+}
+
+// Draw implements Ebiten's Draw method.
+func (g *game) Draw(screen *ebiten.Image) {
+	// draw the UI onto the screen
+	g.ui.Draw(screen)
+}
+
+func loadButtonImage() (*widget.ButtonImage, error) {
+	idleImage := ebiten.NewImage(100, 100)
+	idleImage.Fill(color.RGBA{R: 80, G: 80, B: 80, A: 255})
```

Comment:
You almost never want to use color.RGBA{}. That one expects the alpha to be pre-multiplied. You are probably looking for color.NRGBA{}

Reply:
https://pkg.go.dev/image/color#RGBA
Basically, RGBA expects the alpha to be applied already, so this below would be invalid:
color.RGBA{R:255, G:255: B: 255, A:128} Instead you'd have to do color.RGBA{R:128, G:128: B: 128, A:128} 

UNLESS you use NRGBA where this is valid
color.NRGBA{R:255, G:255: B: 255, A:128}

Reply:
Got it. Thanks.

This helped: [Understanding the Color type in Go](https://dev.to/mokiat/understanding-the-color-type-in-go-1k45)